src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs

```diff
@@ -94,9 +104,72 @@ public void Initialize(IFederationHistory federationHistory, IIdleFederationMemb
             this.nodeStats.RegisterStats(this.AddComponentStats, StatsType.Component, this.GetType().Name, 1200);
 
             this.isInitialized = true;
+
             this.logger.LogDebug("VotingManager initialized.");
         }
 
+        /// <summary> Remove all polls that started on or after the given height.</summary>
+        /// <param name="height">The height to clean polls from.</param>
+        public void DeletePollsAfterHeight(int height)
+        {
+            this.logger.LogInformation($"Cleaning poll data from height {height}.");
+
+            var idsToRemove = new List<int>();
+
+            this.polls = this.pollsRepository.GetAllPolls();
+
+            foreach (Poll poll in this.polls.Where(p => p.PollStartBlockData.Height >= height))
+            {
+                idsToRemove.Add(poll.Id);
+            }
+
```

Comment:
Should we also reset `PollVotedInFavorBlockData` and `PollExecutedBlockData` where these fields contain heights that are >= height?

Reply:
@quantumagi I've decided to, in the follow up PR, to just always reset the federation from before we kicked some of the initial genesis members. I tried testing this with various heights and I did run into problems. So for safety sake I have reverted the code to always rebuild from a set height.

If we ever require the need to rebuild the federation from a given height we will can look at this again.

There is also not really a lot of reduction in processing duration by specifying various "reconstruction" heights.